src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -143,9 +143,6 @@ export class ConnectionDialogService implements IConnectionDialogService {
 				if (filteredKeys && filteredKeys.length > 0) {
 					defaultProvider = filteredKeys[0];
 				}
-				else {
-					defaultProvider = keys[0];
```

Comment:
I'm not clear why this condition can be removed?  How does this work in the case where the MSSQL extension is not installed?

Reply:
i.e. if it's not needed then why is it even impacting this scenario at all?

Reply:
Isn't MSSQL always installed? This function defaults to MSSQL if a provider isn't found.

Reply:
I'm removing the already existing conditional that would pick the first connection regardless for default. So, this could take any provider as the default depending on how they were ordered when being registered.

Reply:
Yes, I see that.  I'm not clear what the behavior is with this change.  What will the default provider be set to in this case?

Reply:
...I looked at the code a little more and it looks like this will run with this change `return defaultProvider || Constants.mssqlProviderName;` which will return `Constants.mssqlProviderName`, which as I mentioned will cause problems if MSSQL extension isn't installed.

Reply:
In regards to is MSSQL always installed, please see previous comment "@abist in case it's not clear, looking up MSSQL may fail if that extension is uninstalled or disabled, for example, if we want to build a PGSQL-only bundle in the future."